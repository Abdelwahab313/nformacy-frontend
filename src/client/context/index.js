import React, { createContext, useContext, useReducer } from 'react';
import {
  SET_ERROR_MESSAGE,
  UPDATE_CLIENTS,
  UPDATE_CURRENT_CLIENT,
  VERIFY_CLIENT,
  VERIFY_CURRENT_CLIENT,
} from './actionTypes';

const ClientContext = createContext();
const defaultValues = {
  clients: [],
  errorMessage: '',
};

const ClientProvider = ({ children, initialValue }) => {
  const { Provider } = ClientContext;
  initialValue = { ...defaultValues, ...initialValue };
  return (
    <Provider value={useReducer(clientReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useClientState = () => {
  const context = useContext(ClientContext);
  if (context === undefined) {
    throw new Error('useClientState must be used within a ClientProvider');
  }
  return context;
};

const clientReducer = (state, action) => {
  switch (action.type) {
    case UPDATE_CLIENTS:
      return {
        ...state,
        clients: action.payload,
      };
    case UPDATE_CURRENT_CLIENT:
      return {
        ...state,
        currentClient: action.payload,
      };
    case VERIFY_CURRENT_CLIENT:
      const currentClient = state.currentClient;
      currentClient.verified = true;
      return {
        ...state,
        currentClient: currentClient,
      };
    case VERIFY_CLIENT:
      const newClients = state.clients.map((client) => {
        if (client[0] === action.payload.uuid) {
          client[6] = true;
          return client;
        } else {
          return client;
        }
      });
      return {
        ...state,
        clients: newClients,
      };
    case SET_ERROR_MESSAGE:
      return {
        ...state,
        errorMessage: action.payload,
      };
  }
};

export { ClientProvider, useClientState };
