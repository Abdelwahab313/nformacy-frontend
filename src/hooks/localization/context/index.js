import React, { createContext, useContext, useReducer } from 'react';

const LocaleContext = createContext();
const DEFAULT_LOCALE = 'en';
const LOCALES = {
  en: 'en',
  ar: 'ar',
};
export const CHANGE_LOCALE = 'CHANGE_LOCALE';

export const useLocaleContext = () => {
  const context = useContext(LocaleContext);
  if (context === undefined)
    throw new Error('useLocale should only be used within locale provider');
  return context;
};

export const LocaleProvider = ({
  children,
  initialLocale = DEFAULT_LOCALE,
}) => {
  const { Provider } = LocaleContext;
  return (
    <Provider value={useReducer(localeReducer, initialLocale)}>{children}</Provider>
  );
};

const localeReducer = (state, action) => {
  if (action.type === CHANGE_LOCALE) {
    return action.payload.locale;
  }
  return LOCALES.en;
};
