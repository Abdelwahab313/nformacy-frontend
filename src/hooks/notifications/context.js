import React, { useContext, useReducer } from 'react';
import { ActionCableProvider } from '../useActionCable';
import { CHANNEL_URL } from '../../settings';
import { produce, current } from 'immer';
import { toast as showToast } from 'react-toastify';

const defaultStates = {
  notifications: [],
  menuOpened: null,
  unread: false,
  unreadCount: 0,
};

const NotificationsContext = React.createContext();

export const notificationActions = {
  notificationReceived: 'notificationReceived',
  menuToggled: 'menuToggled',
  menuClosed: 'menuClosed',
};

const MAX_NOTIFICATIONS = 10;
export const useNotificationsContext = () => {
  const context = useContext(NotificationsContext);
  if (context === undefined) {
    throw new Error(
      'useNotificationsContext must be used within a NotificationsProvider',
    );
  }
  return context;
};
const initContext = (initialValue) => {
  return initialValue || defaultStates;
};

export const NotificationsProvider = ({
  children,
  initialNotifications = [],
  unreadCount = 0,
}) => {
  const { Provider } = NotificationsContext;
  const initialState = {
    ...defaultStates,
    notifications: initialNotifications,
    unread: unreadCount > 0,
    unreadCount,
  };
  return (
    <Provider
      value={useReducer(NotificationsReducer, initialState, initContext)}>
      <ActionCableProvider url={CHANNEL_URL}>{children}</ActionCableProvider>
    </Provider>
  );
};

function Notification(notification) {
  return {
    notificationId: notification.notification_id,
    targetId: notification.target_id,
    messageKey: notification.message_key,
    createdAt: notification.created_at,
    type: notification.type,
    readAt: null,
  };
}

const updateUserInLocalStorage = (currentUser, updatedFields) => {
  localStorage.setItem(
    'user',
    JSON.stringify({ ...currentUser, ...updatedFields }),
  );
};

const NotificationsReducer = (state, action) => {
  switch (action.type) {
    case notificationActions.notificationReceived:
      const currentUser = action.payload.currentUser;
      const receivedNotification = Notification(action.payload.notification);
      showToast(receivedNotification.messageKey, {
        toastId: receivedNotification.notificationId,
      });
      return produce(state, (draftState) => {
        if (draftState.notifications.length === MAX_NOTIFICATIONS) {
          draftState.notifications.pop();
        }
        draftState.notifications.unshift(receivedNotification);
        draftState.unread = true;
        draftState.unreadCount = state.unreadCount + 1;
        updateUserInLocalStorage(currentUser, {
          unreadNotifications: draftState.unreadCount,
          notifications: current(draftState.notifications),
        });
      });
    case notificationActions.menuToggled:
      return produce(state, (draftState) => {
        if (state.menuOpened?.contains(action.payload.target)) {
          draftState.menuOpened = null;
        } else {
          draftState.menuOpened = action.payload.currentTarget;
          draftState.unread = false;
        }
      });
    case notificationActions.menuClosed:
      return produce(state, (draftState) => {
        draftState.menuOpened = null;
      });
    default:
      throw Error(
        `Action: ${action.type} is not defined in notifications reducer`,
      );
  }
};
