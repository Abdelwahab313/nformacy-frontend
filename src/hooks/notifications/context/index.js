import React, { useContext, useReducer } from 'react';
import { ActionCableProvider } from '../useActionCable';
import { CHANNEL_URL } from '../../../settings';
import {
  clearToast,
  closeMenu,
  loadNotifications,
  receiveNotification,
  toggleMenu,
  visitNotification,
} from './actions';

const defaultStates = {
  notifications: [],
  menuOpened: null,
  unread: false,
  unreadCount: 0,
};

const NotificationsContext = React.createContext();

export const notificationActions = {
  notificationsLoaded: 'notificationsLoaded',
  notificationReceived: 'notificationReceived',
  notificationVisited: 'notificationVisited',
  menuToggled: 'menuToggled',
  menuClosed: 'menuClosed',
  toastCleared: 'toastCleared',
};

export const useNotificationsContext = () => {
  const context = useContext(NotificationsContext);
  if (context === undefined) {
    throw new Error(
      'useNotificationsContext must be used within a NotificationsProvider',
    );
  }
  return context;
};
const initContext = (initialValue) => {
  return initialValue || defaultStates;
};

export const NotificationsProvider = ({ children }) => {
  const { Provider } = NotificationsContext;
  const initialState = {
    ...defaultStates,
    notifications: [],
    unreadCount: 0,
  };
  return (
    <Provider
      value={useReducer(NotificationsReducer, initialState, initContext)}>
      <ActionCableProvider url={CHANNEL_URL}>{children}</ActionCableProvider>
    </Provider>
  );
};

const NotificationsReducer = (state, action) => {
  switch (action.type) {
    case notificationActions.notificationsLoaded:
      return loadNotifications(state, action);
    case notificationActions.notificationReceived:
      return receiveNotification(action, state);
    case notificationActions.menuToggled:
      return toggleMenu(state, action);
    case notificationActions.menuClosed:
      return closeMenu(state);
    case notificationActions.notificationVisited:
      return visitNotification(state, action);
    case notificationActions.toastCleared:
      return clearToast(state);
    default:
      throw Error(
        `Action: ${action.type} is not defined in notifications reducer`,
      );
  }
};
