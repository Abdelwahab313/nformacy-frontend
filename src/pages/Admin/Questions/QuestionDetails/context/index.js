import React, { createContext, useContext, useReducer } from 'react';
import QuestionActionTypes from './actionTypes';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      const updatedQuestion = { ...state.questionDetails, ...action.payload };
      return { ...state, questionDetails: updatedQuestion };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext, QuestionProvider };
