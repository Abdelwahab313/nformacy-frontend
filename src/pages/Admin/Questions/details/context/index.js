import React, { createContext, useContext, useReducer } from 'react';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

export const QuestionActionTypes = {
  UPDATE_QUESTION: 'UPDATE_QUESTION',
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      return { ...state, questionDetails: action.payload };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext as useAuth, QuestionProvider as AuthProvider };
