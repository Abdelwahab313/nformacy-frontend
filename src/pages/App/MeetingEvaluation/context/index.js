import React, { createContext, useContext, useReducer } from 'react';
// import CallEvaluationActionTypes from './actionTypes';

const defaultValues = {
  comment: '',
  ratingEvaluations: {}
};

const CallEvaluationContext = createContext();

const CallEvaluationProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = CallEvaluationContext;
  return (
    <Provider value={useReducer(CallEvaluationReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useCallEvaluationContext = () => {
  const context = useContext(CallEvaluationContext);
  if (context === undefined) {
    throw new Error('useCallEvaluationContext must be used within a CallEvaluationContext');
  }
  return context;
};

const CallEvaluationReducer = (state, action) => {
  switch (action.type) {

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useCallEvaluationContext, CallEvaluationProvider };
