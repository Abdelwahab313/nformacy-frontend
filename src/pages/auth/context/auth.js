import React, { createContext, useContext, useReducer } from 'react';

const defaultValues = {
  currentUser: '',
};
const AuthContext = createContext();

const AuthProvider = ({ children, initialValue = defaultValues }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = AuthContext;
  return (
    <Provider value={useReducer(AuthReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useAuth = () => {
  const context = useContext(AuthContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within a AuthProvider');
  }
  return context;
};

export const AuthActionTypes = {
  UPDATE_CURRENT_USER: 'UPDATE_CURRENT_USER',
  LOGOUT_USER: 'LOGOUT_USER',
};

const AuthReducer = (state, action) => {
  switch (action.type) {

    case AuthActionTypes.UPDATE_CURRENT_USER:
      localStorage.setItem('user', JSON.stringify(action.payload));
      return { ...state, currentUser: action.payload };

    case AuthActionTypes.LOGOUT_USER:
      localStorage.removeItem('user');
      return { ...state, currentUser: '' };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};


export { useAuth, AuthProvider };
