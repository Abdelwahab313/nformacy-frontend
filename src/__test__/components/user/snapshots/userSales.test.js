import { shallow } from 'enzyme';
import * as AuthContextMoudle from '../../../../context/auth';
import { AuthContext } from '../../../../context/auth';
import React from 'react';
import UserSalesScreen from '../../../../components/user/userDetail/UserSale';

function mockAuthObject() {
  jest.spyOn(AuthContextMoudle, 'useAuth').mockImplementation(() => ({
    authTokens: '',
  }));
}

const testUsersList = [
  {
    uuid: 'randomstring',
    first_name: 'first name',
    last_name: 'second name',
    phone_number: '01099234802',
    national_id: '29505050103981',
    username: 'omarmmali',
    date_joined: '05/05/1995',
  },
];

it('renders correctly', () => {
  mockAuthObject();

  const container = shallow(
    <AuthContext.Provider>
      <UserSalesScreen user_uuid={testUsersList[0].uuid} />
    </AuthContext.Provider>,
  );
  const tree = container.html();
  expect(tree).toMatchSnapshot();
});
