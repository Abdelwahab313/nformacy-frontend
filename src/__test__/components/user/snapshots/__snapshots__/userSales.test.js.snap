// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div class=\\"makeStyles-root-1\\" dir=\\"rtl\\"><div class=\\"MuiGrid-root tableContainer makeStyles-tableContainerFW-7\\"><div class=\\"MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded\\" style=\\"position:relative\\"><div class=\\"MuiGrid-root MuiGrid-grid-lg-12\\"><div class=\\"filterContainer\\"><div class=\\"MuiGrid-root filterTitle MuiGrid-grid-lg-3\\"><label>تصفية النتائج:</label></div><div class=\\"MuiGrid-root filterInputContainer MuiGrid-grid-lg-9\\"><div><label>من</label><div class=\\"react-datepicker-wrapper\\"><div class=\\"react-datepicker__input-container\\"><input type=\\"text\\" value=\\"11/26/2019\\" class=\\"\\"/></div></div></div><div><label>الي</label><div class=\\"react-datepicker-wrapper\\"><div class=\\"react-datepicker__input-container\\"><input type=\\"text\\" value=\\"11/26/2019\\" class=\\"\\"/></div></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">عرض</span></button></div></div><div class=\\"MuiToolbar-root MuiToolbar-regular MTableToolbar-root-174 MuiToolbar-gutters\\"><div class=\\"MTableToolbar-title-178\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">المبيعات</h6></div><div class=\\"MTableToolbar-spacer-176\\"></div><div class=\\"MuiFormControl-root MuiTextField-root MTableToolbar-searchField-179\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionStart\\"><span class=\\"material-icons MuiIcon-root MuiIcon-fontSizeSmall\\" aria-hidden=\\"true\\" title=\\"بحث\\">search</span></div><input type=\\"text\\" aria-invalid=\\"false\\" placeholder=\\"بحث\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd\\"/><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root MuiIcon-fontSizeSmall\\" aria-hidden=\\"true\\">clear</span></span></button></div></div></div><div class=\\"MTableToolbar-actions-177\\"><div><div><span></span></div></div></div></div></div><div style=\\"overflow-x:auto\\"><div><div style=\\"overflow-y:auto\\"><table class=\\"MuiTable-root\\"><thead class=\\"MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-289 MuiTableCell-paddingCheckbox\\" scope=\\"col\\" style=\\"z-index:0;text-align:center\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root Mui-disabled\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"true\\">عمليات<svg class=\\"MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-289 MuiTableCell-alignLeft\\" scope=\\"col\\" style=\\"z-index:0\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div data-react-beautiful-dnd-draggable=\\"0\\" tabindex=\\"0\\" data-react-beautiful-dnd-drag-handle=\\"0\\" aria-roledescription=\\"Draggable item. Press space bar to lift\\" draggable=\\"false\\">اسم العميل</div><span class=\\"material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc\\" aria-hidden=\\"true\\">arrow_downward</span></span></th><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-289 MuiTableCell-alignLeft\\" scope=\\"col\\" style=\\"z-index:0\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div data-react-beautiful-dnd-draggable=\\"0\\" tabindex=\\"0\\" data-react-beautiful-dnd-drag-handle=\\"0\\" aria-roledescription=\\"Draggable item. Press space bar to lift\\" draggable=\\"false\\">الحساب الكلي</div><span class=\\"material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc\\" aria-hidden=\\"true\\">arrow_downward</span></span></th><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-289 MuiTableCell-alignLeft\\" scope=\\"col\\" style=\\"z-index:0\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div data-react-beautiful-dnd-draggable=\\"0\\" tabindex=\\"0\\" data-react-beautiful-dnd-drag-handle=\\"0\\" aria-roledescription=\\"Draggable item. Press space bar to lift\\" draggable=\\"false\\">التاريخ</div><span class=\\"material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc\\" aria-hidden=\\"true\\">arrow_downward</span></span></th></tr></thead><tbody class=\\"MuiTableBody-root\\"><tr class=\\"MuiTableRow-root\\" style=\\"height:245px\\"><td class=\\"MuiTableCell-root MuiTableCell-body\\" style=\\"padding-top:0;padding-bottom:0;text-align:center\\" colSpan=\\"4\\">لا يوجد بيانات لعرضها</td></tr></tbody></table></div></div></div><table class=\\"MuiTable-root\\"><tfoot class=\\"MuiTableFooter-root\\" style=\\"display:grid\\"><tr class=\\"MuiTableRow-root MuiTableRow-footer\\"><td class=\\"MuiTableCell-root MuiTableCell-footer MuiTablePagination-root Component-paginationRoot-112\\" colSpan=\\"3\\" style=\\"float:right;overflow-x:auto\\" icons=\\"[object Object]\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar Component-paginationToolbar-113 MuiToolbar-gutters\\"><div class=\\"MuiTablePagination-spacer\\"></div><p class=\\"MuiTypography-root MuiTablePagination-caption Component-paginationCaption-114 MuiTypography-body2 MuiTypography-colorInherit\\">Rows per page:</p><div class=\\"MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot Component-paginationSelectRoot-115\\"><div class=\\"MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input MuiInputBase-inputSelect\\" tabindex=\\"0\\" role=\\"button\\" aria-labelledby=\\" \\" aria-haspopup=\\"listbox\\"><div style=\\"padding:0px 5px\\">5 مبيعات </div></div><input type=\\"hidden\\" value=\\"5\\"/><svg class=\\"MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div><p class=\\"MuiTypography-root MuiTablePagination-caption Component-paginationCaption-114 MuiTypography-body2 MuiTypography-colorInherit\\">0-0 من 0</p><div class=\\"MTablePaginationInner-root-348\\"><span title=\\"الصفحه الاولى\\" class=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"الصفحه الاولى\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">first_page</span></span></button></span><span title=\\"الصفحه السابقه\\" class=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"الصفحه السابقه\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">chevron_left</span></span></button></span><span class=\\"MuiTypography-root MuiTypography-caption\\" style=\\"flex:1;text-align:center;align-self:center;flex-basis:inherit\\">1-0 من 0</span><span title=\\"الصفحه التاليه\\" class=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"الصفحه التاليه\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">chevron_right</span></span></button></span><span title=\\"الصفحه الأخيره\\" class=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"الصفحه الأخيره\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">last_page</span></span></button></span></div></div></td></tr></tfoot></table></div></div></div>"`;
